### PR TITLE
Improves player retention by leveraging existing assets to incentivize higher average playetime

### PR DESCRIPTION
**Pull request by MrDoomBringer**
image
## About The Pull Request
encourages players to play more

## Why It's Good For The Game
playing good, see?

## Changelog
:cl: MrDoomBringer
add: You should seriously consider playing for over five thousand hours, okay?
:cl:

### CODE REVIEW

File: code/datums/skills/_skill.dm [hunk 1]
```diff
@@ -69,7 +69,7 @@ GLOBAL_LIST_INIT(skill_types, subtypesof(/datum/skill))
 	if (!ispath(skill_cape_path))	
 		to_chat(mind.current, "<span class='nicegreen'>My legendary [name] skill is quite impressive, though it seems the Professional [title] Association doesn't have any status symbols to commemorate my abilities with. I should let Centcom know of this travesty, maybe they can do something about it.</span>")
 		return
-	if (LAZYFIND(mind.skills_rewarded, src.type))	
+	if (LAZYFIND(mind.skills_rewarded, src.type) || (istype(type, /datum/skill/playing) && mind.current.client?.prefs.playtime_reward_cloak))	//The playing skill can get it's reward claok from the preferences menu
```
Comment (ATH1909) [inline code/datums/skills/_skill.dm:72]:
```suggestion
	if (LAZYFIND(mind.skills_rewarded, src.type) || (istype(type, /datum/skill/playing) && mind.current.client?.prefs.playtime_reward_cloak))	//The playing skill can get its reward claok from the preferences menu
```

**Comment by ATH1909** [comment]
Does this count the number of hours you've played prior to the creation of this PR?

If I've racked up 1,000 hours on Bagil, do I still have to rack up 1,000 hours on Sybil to be able to use the playing skill cape on Sybil?

**Comment by Iamgoofball** [comment]
yeah, this should ask the DB for played hours

**Comment by MrDoomBringer** [comment]
this does ask the DB for played hours

**Comment by MrDoomBringer** [comment]
@Iamgoofball b-but this PR does poll the DB for played hours

**Comment by MrDoomBringer** [comment]
@ATH1909 it tracks hours across servers

**Comment by Fikou** [comment]
lexia cape lexia cape

**Comment by ArcaneMusic** [comment]
Oof ouch my immersion

**Comment by trollbreeder** [comment]
call it long time employment, not playing for a long time

**Comment by IndieanaJones** [comment]
I don't like this.  Seems to be an enabler for giving "veterans" preferential treatment.

**Comment by Qustinnus** [comment]
this shouldn't be a skill. this is an extremely snowflake implementation

**Comment by necromanceranne** [comment]
I feel bad I'm a 1/4th of the way there....

File: code/__DEFINES/misc.dm [hunk 2]
```diff
@@ -492,6 +492,7 @@ GLOBAL_LIST_INIT(pda_styles, sortList(list(MONO, VT, ORBITRON, SHARE)))
 
 // Play time / EXP
 #define PLAYTIME_HARDCORE_RANDOM 120
+#define PLAYTIME_VETERAN 10000 HOURS
```
Comment (Qustinnus) [inline code/__DEFINES/misc.dm:495]: this is in minutes so this is 6000000 hours

**Comment by MrDoomBringer** [comment]
its no longer a skill and is now more IC

File: code/__DEFINES/misc.dm (hunk 2 above)
Comment (MrDoomBringer) [reply code/__DEFINES/misc.dm]: ur 6 billion hours or whatever

Comment (MrDoomBringer) [reply code/__DEFINES/misc.dm]: ok so? just play more casual

**Comment by wesoda25** [comment]
Doom you are being an enabler 😭🐓

**Comment by Mickyan** [comment]
Being perfectly aware of how unlikely anyone is to ever get this, the problem with joke items like this is that it may take one, two years or even more, someone will use this as a precedent to add more and more obnoxious items as persistent rewards.

And then one day we need to have a giant debate about how people had to work for their obnoxious rewards and you can't just take them away when every person on the server looks like a christmas tree.

**Comment by MrDoomBringer** [comment]
do you want me to up it to 6million hours or what

**Comment by Qustinnus** [comment]
just keep raising the hours required problem solved

**Comment by wesoda25** [comment]
10,000 hours is well over a year, perhaps we should just automatically ban them for wasting so much of their fucking life.

**Comment by optimumtact** [comment]
I requested this :+1:

**Comment by MrDoomBringer** [comment]
image

**Comment by Doosteroni** [comment]
This shit looks good, but if someone get's 10k hours I would bankroll a hitman not give them a sweet ass cape. Now here's a good idea. Give this to people who master all other the skills. That's it

**Comment by cacogen** [comment]
finally something to show for this but i bet what it's recorded isn't enough

**Comment by Cyberboss** [comment]
This is literally a waste of code, lmao. Only people that will get this are admins that use +DEBUG to forge querys to update their playtime.

Feel free to PR it as a reward for something else.

**Comment by optimumtact** [comment]
There are literally players with 10,000 hours of playtime

**Comment by optimumtact** [comment]
set it back to 10,000 hours you dent head

**Comment by Qustinnus** [comment]
playtime is in minutes

**Comment by optimumtact** [comment]
you saw nothing

**Comment by Cyberboss** [comment]
>There are players with over 1.25 years of playtime

Why do I maintain this game?

**Comment by optimumtact** [comment]
You have probably sunk about 10,000 hours into maintaining and developing it you know?

**Comment by Qustinnus** [comment]
and most of us have spent 10000 hours bikeshedding

**Comment by Cyberboss** [comment]
Very debatable, that's more than 7 hours a day, every day, for four years.

I work full time

**Comment by optimumtact** [comment]
you do now

**Comment by Cyberboss** [comment]
https://www.theolognion.com/open-source-maintainer-apologizes-for-having-a-day-time-job/

**Review by Cyberboss — CHANGES_REQUESTED** [review]
Discussed in Discord, lower it to 5000 hours please.

File: code/__DEFINES/misc.dm [hunk 3]
```diff
@@ -492,6 +492,7 @@ GLOBAL_LIST_INIT(pda_styles, sortList(list(MONO, VT, ORBITRON, SHARE)))
 
 // Play time / EXP
 #define PLAYTIME_HARDCORE_RANDOM 120
+#define PLAYTIME_VETERAN 600000 //Playtime is tracked in minutes. 600,000 minutes = 10,000 hours
```
Comment (MrDoomBringer) [inline code/__DEFINES/misc.dm:495]:
```suggestion
#define PLAYTIME_VETERAN 300000 //Playtime is tracked in minutes. 300,000 minutes = 5,000 hours
```
lower the playtime to 5000 hours please

Comment (MrDoomBringer) [reply code/__DEFINES/misc.dm:495]: a splendid idea, i'll push the commit suggestion button post haste

**Review by Cyberboss — DISMISSED** [review]

**Review by Cyberboss — DISMISSED** [review]
Fuck you fucking dumb ass fucking bastard bot

**Comment by MrDoomBringer** [comment]
S T A L E

**Review by Cyberboss — APPROVED** [review]

**Comment by Cyberboss** [comment]
@MrDoomBringer your changelog didn't gen because you forgot the `/`. You can open another PR for it

**Comment by IndieanaJones** [comment]
This sucks.

**Comment by optimumtact** [comment]
loser with 3000 hours detected

**Pull request merged**